contracts/commands/BasicBuyCommand.sol

```diff
@@ -96,11 +77,28 @@ contract BasicBuyCommand is BaseMPACommand, Ownable {
     {
         BasicBuyTriggerData memory decoded = decode(triggerData);
 
-        (, uint256 nextCollRatio, uint256 nextPrice, ) = getVaultAndMarketInfo(cdpId);
+        (
+            uint256 collRatio,
+            uint256 nextCollRatio,
+            uint256 currPrice,
+            uint256 nextPrice,
+            bytes32 ilk
+        ) = getVaultAndMarketInfo(cdpId);
+
+        SpotterLike spot = SpotterLike(serviceRegistry.getRegisteredService(MCD_SPOT_KEY));
+        (, uint256 liquidationRatio) = spot.ilks(ilk);
+
         return
             nextCollRatio != 0 &&
             nextCollRatio >= decoded.execCollRatio.wad() &&
-            nextPrice <= decoded.maxBuyPrice;
+            nextPrice <= decoded.maxBuyPrice &&
+            collRatio >
+            liquidationRatio
+                .radToWad()
+                .add(nextCollRatio)
+                .sub(decoded.targetCollRatio.wad())
+                .mul(currPrice)
+                .div(nextPrice);
```

Comment:
parameters that passes validation and do not lead to correct execution

nextPrice 2000
currPrice 1499

liqRatio 150

currentCollRatio = 152
currentCollRatioAtNextPrice = 202.8

executionCollRatio = 202
targetCollRatio = 195